components/cards/ClaimAction.tsx

```diff
@@ -13,78 +21,172 @@ export type ClaimActionProps = SidebarProps & {
+        `displayNewForSalePrice is invalid: ${displayNewForSalePrice}`
+      );
+    }
+
+    console.log(actionData);
+
+    let claimData;
+
+    let actionDataToPassInUserData;
+
+    let userData;
+
+    let txn;
+
+    const filter = claimerContract.filters.ParcelClaimed(null, null);
+    const parcels = await claimerContract.queryFilter(filter);
+    const existingParcels = parcels.length > 0;
+    console.log("existing parcels: ", parcels);
+
+    const sf = await sfInstance(NETWORK_ID, provider);
+
+    const ethx = await sf.loadSuperToken(paymentTokenAddress);
+
+    const approveOperation = await ethx.approve({
+      receiver: auctionSuperApp.address,
+      amount: ethers.utils.parseEther(displayNewForSalePrice).toString(),
+    });
+
+    const newFlowRate = ethers.utils
+      .parseEther(displayNewForSalePrice)
+      .div(SECONDS_IN_YEAR);
+
+    const signer = provider.getSigner() as any;
+
+    if (existingParcels) {
+      // update an exisiting flow
+      console.log("Updting an exisiting flow: ");
+      const existingLicenseID = parcels[0].args[0].toString();
+
+      claimData = ethers.utils.defaultAbiCoder.encode(
+        ["uint256"],
+        [existingLicenseID]
+      );
+
+      actionDataToPassInUserData = ethers.utils.defaultAbiCoder.encode(
+        ["uint256", "bytes"],
+        [displayNewForSalePrice, claimData]
+      );
+
+      userData = ethers.utils.defaultAbiCoder.encode(
+        ["uint8", "bytes"],
+        [Action.CLAIM, actionDataToPassInUserData]
+      );
+
+      const existingFlow = await sf.cfaV1.getFlow({
+        superToken: paymentTokenAddress,
+        sender: account,
+        receiver: auctionSuperApp.address,
+        providerOrSigner: provider as any,
+      });
+
+      const updateFlowOperation = await sf.cfaV1.updateFlow({
+        flowRate: BigNumber.from(existingFlow.flowRate)
+          .add(newFlowRate)
+          .toString(),
+        receiver: auctionSuperApp.address,
+        superToken: paymentTokenAddress,
+        userData,
+      });
+
+      txn = await sf
+        .batchCall([approveOperation, updateFlowOperation])
+        .exec(signer);
+
+      console.log(txn);
+    } else {
+      // create a new flow
+      console.log("Creating a new flow: ");
+      claimData = ethers.utils.defaultAbiCoder.encode(
+        ["uint64", "uint256[]"],
+        [BigNumber.from(baseCoord.toString(10)), path]
+      );
+
+      userData = ethers.utils.defaultAbiCoder.encode(
+        ["uint8", "bytes"],
+        [Action.CLAIM, claimData]
+      );
+
+      const flowRate = newFlowRate.toString();
+
+      console.log("flowRate: ", flowRate);
+
+      const createFlowOperation = await sf.cfaV1.createFlow({
+        flowRate,
+        receiver: auctionSuperApp.address,
+        superToken: paymentTokenAddress,
+        userData,
+      });
+
+      txn = await sf
+        .batchCall([approveOperation, createFlowOperation])
+        .exec(signer);
+
+      console.log(txn);
```

Comment:
@codynhat 

Claim flow explanation [1/2]:

> If this is the first parcel a user is claiming, the flow should be created

Reply:
Looks like you are missing `actionDataToPassInUserData` in the create flow case.